spec/solve_spec.rb

```diff
@@ -0,0 +1,24 @@
+require_relative './spec_helper'
+
+describe Solver do
+  it 'test for the solver factorial method' do
+    solve = Solver.new
+    expect(solve.factorial(5)).to be_kind_of(Integer)
+    expect(solve.factorial(5)).to eq(120)
+    expect(solve.factorial(0)).to eq(1)
+    expect(solve.factorial(1)).to eq(1)
```

Comment:
- [ ] Please add a test case for the factorial of a `Negative number` which raises an exception as an expectation to cover all test cases.